src/locales/strings.test.js

```diff
@@ -0,0 +1,48 @@
+import { getTranslator } from './strings.js'
+
+describe('TranslateFn', () => {
+    it('uses the provided language', () => {
+        const en = getTranslator({ language: 'en' })
+        expect(en('hello')).toBe('Hello world')
+
+        const xa = getTranslator({ language: 'xa' })
+        expect(xa('hello')).toBe('H33ll00 wºrrld')
+    })
+
+    it('defaults to en for unsupported languages', () => {
+    // Silence warnings, since we expect one here.
+        jest.spyOn(console, 'warn').mockImplementation(() => {})
+        const t = getTranslator({ language: 'does-not-exist' })
+        expect(t('hello')).toBe('Hello world')
+    })
+
+    it('reads language just in time', () => {
+        const settings = { language: 'does-not-exist' }
+        // Get a translator, but don't call it until after setting a valid language.
+        const t = getTranslator(settings)
+        settings.language = 'en'
+        expect(t('hello')).toBe('Hello world')
```

Comment:
Is this testing anything? If you pass `does-not-exist`, doesn't it fallback to `en` anyway? 🤔 Maybe line 23 should read `settings.language = 'xa'`?

Reply:
D'oh, good catch!